script.js

```diff
@@ -1 +1,152 @@
+    todo.appendChild(delBtn);
+    todo.appendChild(completeBtn);
+    todo.id=todoId;
+    todo.classList.add("todoList");
+
+    todoList.appendChild(todo);
+
+    const todoObj = {
+        text : text,
+        id: todoId,
+    };
+
+    todos.push(todoObj);
+    saveToLocal('todo');
+}
+
+//todo 리스트에서 할일 끝내면 complete 리스트에 할일을 text 기반으로 추가 
+function addComplete(text){
+    const completeTodo = document.createElement("li");
+    const cancelBtn = document.createElement("button");
+    const delBtn = document.createElement("button");
+    const span = document.createElement("span");
+
+    cancelBtn.classList.add("cancelimg");
+    delBtn.classList.add("deleteimg");
+
+    cancelBtn.addEventListener("click", deleteCompleteTodo);  //끝낸 일에서 다시 할 일로 올림
+    delBtn.addEventListener("click", deleteCompleteTodo);   //끝낸 일 삭제
+    span.innerText = text;
+    const completeID = done.length+1;
+
+    completeTodo.appendChild(delBtn);
+    completeTodo.appendChild(cancelBtn);
+    completeTodo.appendChild(span);
+    completeTodo.classList.add("completeList");
+    completeTodo.id = completeID;               //이걸 추가해야 나중에 event처리할 때 id 를 이용해서 구분이 가능..! 오류 잡으려고 30분 넘게 걸렸으니까 제발,,,까먹지 말아야지...
+
+    completeList.appendChild(completeTodo);
+
+    const doneObj = {
+        text: text,
+        id : completeID,
+    };
+    done.push(doneObj);
+    saveToLocal('done');
+}
+
+//todolist 에서 할일 삭제, 이벤트 기반
+function deleteTodo(event){
+    const deleteNode =  event.target.parentNode; //버튼의 부모노드를 반환해줌, 여기서 id를 찾을 수 있음!
+    console.dir(deleteNode)
+    todoList.removeChild(deleteNode);
+    const cleanTodo = todos.filter(function(toDo){
+        return toDo.id !== parseInt(deleteNode.id);       //deleteNode.id 의 값은 string, 따라서 형변환 필요
+    });   //filter는 배열의 모든 아이템을 통해 함수를 실행, 함수 값이 true 인 아이템들로만 새로운 배열을 생성
+
+    todos=cleanTodo;
+    saveToLocal('todo');          //실수 했던 부분, 로컬스토리지에 저장하려면 이걸 꼭 썼어야 함,,
+}
+
+//할일 끝내면 todolist에서 할일 삭제, completelist에 할일 추가, 이벤트 기반
+function completeTodo(event){
+    const completeNode = event.target.parentNode;
+    deleteTodo(event);
+    addComplete(completeNode.textContent);    //completeNode.outerText 이거를 대신 써도 되는듯
+}
+
+//completelist 의 할 일 삭제
+function deleteCompleteTodo(event){
+    const delCompleteNode = event.target.parentNode;
+    console.dir(delCompleteNode)
+    if (event.target.className==="cancelimg"){   //삭제가 아니라 할 일로 넘겨야하는 경우
+        addToDo(delCompleteNode.textContent);
+    }
+    completeList.removeChild(delCompleteNode);
+    const cleanDone = done.filter(function(toDo){
+        return toDo.id !== parseInt(delCompleteNode.id);       //deleteNode.id 의 값은 string, 따라서 형변환 필요
+    });
+
+    done = cleanDone;
+    saveToLocal('done');
+}
+
+//로컬스토리지에 할 일 저장 (todolist만)
+function saveToLocal(status){
+    if (status === 'todo'){
+        localStorage.setItem(TODOS_LS,JSON.stringify(todos));    //js 객체를 String으로 바꿔서 로컬 스토리지에 저장
+    }else{
+        localStorage.setItem(COM_LS,JSON.stringify(done));
+    }
```

Comment:
switch 문을 사용해보는 건 어떨까요??????!!!!

todo의 status를 enum처럼 생각하고
```suggestion
    switch (status) {
        case 'TODO':
            localStorage.setItem(TODOS_LS,JSON.stringify(todos));    //js 객체를 String으로 바꿔서 로컬 스토리지에 저장
            break;
        case 'COMPLETED':
            localStorage.setItem(COM_LS,JSON.stringify(done));
            break;
        default:
            throw new Error();
    }
```